### PR TITLE
Allow ProcessRunnerTest based tests to take extra arguments

### PR DESCRIPTION
**Pull request by nober97**
## Bug / Requirement Description
Some tests can take extra user parameters. However, they can't be added to the executable. The goal is to let ProcessRunnerTest based tests to pass user arguments to the executable in list and in execute mode as well.

## Solution description
ProcessRunnerTest's config was extended with two parameters (pre_args and post_args) that meant to be prepended and appended to the list of args to the test runnable in list_command and test_command methods. The original methods were made to private members of the class.

## Checklist:
- [x] Test
- [x] News fragment present for release notes
- [x] MS info leakage check

### CODE REVIEW

File: testplan/testing/base.py [hunk 1]
```diff
@@ -488,11 +490,19 @@ class ProcessRunnerTest(Test):
...
             proc_env.update(options["proc_env"])
         options["proc_env"] = proc_env
+        if pre_args:
+            options["pre_args"] = pre_args
+        else:
+            options["pre_args"] = []
+        if post_args:
+            options["post_args"] = post_args
+        else:
+            options["post_args"] = []
```
Comment (Pyifan) [inline testplan/testing/base.py:505]: If we remove pre_args & post_args from the parameter list, then they gets to **options, and will be updated to self.cfg.pre_args/post_args automatically when Entity class's __init__ is  called. Then we don't need these few lines.

File: testplan/testing/base.py [hunk 2]
```diff
@@ -529,7 +539,13 @@ def test_command(self):
...
         :rtype: ``list`` of ``str``
         """
-        return [self.cfg.binary]
+        if self.cfg.pre_args:
+            cmd = [*self.cfg.pre_args, self.cfg.binary]
+        else:
+            cmd = [self.cfg.binary]
+        if self.cfg.post_args:
+            cmd.extend(self.cfg.post_args)
+        return cmd
```
Comment (Pyifan) [inline testplan/testing/base.py:548]: Similar logic is repeated in derived class, and I think we could avoid this. One possible way is to move the original logic to test_command to _test_command (and inheritances will override accordingly), and in base class test_command, add pre/post_args (and inheritances do not need to override).

**Review by Pyifan — CHANGES_REQUESTED** [review]
We also need news fragment, and a test (or an example would also do)

File: testplan/testing/base.py [hunk 3]
```diff
@@ -437,6 +437,8 @@ def get_options(cls):
                 None, float, int, Use(parse_duration)
             ),
             ConfigOption("ignore_exit_codes", default=[]): [int],
+            ConfigOption("pre_args", default=[]): list,
+            ConfigOption("post_args", default=[]): list,
```
[the remark this answers is not in the document]
Comment (nober97) [reply testplan/testing/base.py:441]: Solved

File: testplan/testing/base.py [hunk 4]
```diff
@@ -488,11 +490,19 @@ class ProcessRunnerTest(Test):
     _VERIFICATION_TESTCASE_NAME = "ExitCodeCheck"
     _MAX_RETAINED_LOG_SIZE = 4096
 
-    def __init__(self, **options):
+    def __init__(self, pre_args=None, post_args=None, **options):
```
[the remark this answers is not in the document]
Comment (nober97) [reply testplan/testing/base.py:493]: Removed.

File: testplan/testing/base.py (hunk 1 above)
Comment (nober97) [reply testplan/testing/base.py:505]: Removed.

**Review by Pyifan — APPROVED** [review]

**Pull request merged**